### ISSUE TITLE
Missing assessment items in installers

### ISSUE THREAD

**Issue by EdDixon** [learningequality/ka-lite#4141]
In all installers tested over the past several weeks a large number of assessment items seem to be missing. I beleve these to be the ones contained in the assessments.zip file and I see an error about a magic number when the kalite command attempts to decompress the file when run manually on windows after installation when using the "kalite manage setup" command "BadZipfile: Bad magic number for central directory" which I feel maybe the cause? Full traceback below:

Microsoft Windows XP [Version 5.1.2600]
(C) Copyright 1985-2001 Microsoft Corp.

C:\Documents and Settings\IEUser>kalite
C:\Python27\Scripts\kalite:98: RuntimeWarning: Could not determine Python in you
r path, defaulting to C:\Python27\python.exe
  warnings.warn("Could not determine Python in your path, defaulting to {0:s}".f
ormat(python_executable), RuntimeWarning)
Usage:
  kalite start [--foreground] [options] [DJANGO_OPTIONS ...]
  kalite stop [options] [DJANGO_OPTIONS ...]
  kalite restart [options] [DJANGO_OPTIONS ...]
  kalite status [options]
  kalite shell [options] [DJANGO_OPTIONS ...]
  kalite test [options] [DJANGO_OPTIONS ...]
  kalite manage [options] COMMAND [DJANGO_OPTIONS ...]
  kalite -h | --help
  kalite --version

C:\Documents and Settings\IEUser>kalite manage setup
C:\Python27\Scripts\kalite:98: RuntimeWarning: Could not determine Python in you
r path, defaulting to C:\Python27\python.exe
  warnings.warn("Could not determine Python in your path, defaulting to {0:s}".f
ormat(python_executable), RuntimeWarning)

   _   __  ___    _     _ _
  | | / / / _ \  | |   (_) |
  | |/ / / /_\ \ | |    _| |_ ___
  |    \ |  _  | | |   | | __/ _ \
  | |\  \| | | | | |___| | ||  __/
  \_| \_/\_| |_/ \_____/_|\__\___|

https://learningequality.org/ka-lite/

         version 0.14.0

--------------------------------------------------------------------------------

This script will configure the database and prepare it for use.
--------------------------------------------------------------------------------

Press [enter] to continue...
-------------------------------------------------------------------
WARNING: Database file already exists!
-------------------------------------------------------------------
Keep database file and upgrade to KA Lite version 0.14.0? (yes or no) yes
Upgrading database to KA Lite version 0.14.0
Cleaning *pyc files (if writable) from: C:\Program Files\KA Lite\ka-lite\dist-pa
ckages
Cleaning *pyc files (if writable) from: C:\Program Files\KA Lite\ka-lite\python-
packages
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)
Running migrations for securesync:
- Nothing to migrate.
 - Loading initial data for securesync.
Installed 0 object(s) from 0 fixture(s)
Running migrations for config:
- Nothing to migrate.
 - Loading initial data for config.
Installed 0 object(s) from 0 fixture(s)
Running migrations for chronograph:
- Nothing to migrate.
 - Loading initial data for chronograph.
Installed 0 object(s) from 0 fixture(s)
Running migrations for main:
- Nothing to migrate.
 - Loading initial data for main.
Installed 0 object(s) from 0 fixture(s)
Running migrations for playlist:
- Nothing to migrate.
 - Loading initial data for playlist.
Installed 0 object(s) from 0 fixture(s)
Running migrations for updates:
- Nothing to migrate.
 - Loading initial data for updates.
Installed 0 object(s) from 0 fixture(s)
Running migrations for facility:
- Nothing to migrate.
 - Loading initial data for facility.
Installed 0 object(s) from 0 fixture(s)
Running migrations for student_testing:
- Nothing to migrate.
 - Loading initial data for student_testing.
Installed 0 object(s) from 0 fixture(s)
Running migrations for store:
- Nothing to migrate.
 - Loading initial data for store.
Installed 0 object(s) from 0 fixture(s)
Running migrations for i18n:
- Nothing to migrate.
 - Loading initial data for i18n.
Installed 0 object(s) from 0 fixture(s)
Running migrations for tastypie:
- Nothing to migrate.
 - Loading initial data for tastypie.
Installed 0 object(s) from 0 fixture(s)
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)

Starting in version 0.13, you will need an assessment items package in order to
access many of the available exercises.
If you have an internet connection, you can download the needed package. Warning
: this may take a long time!
If you have already downloaded the assessment items package, you can specify the
 file in the next step.
Otherwise, we will download it from https://learningequality.org/downloads/ka-li
te/0.14/content/assessment.zip.
Do you wish to download the assessment items package now? (yes or no) yes
Downloading assessment item data from a remote server. Please be patient; this f
ile is big, so this may take some time...
Downloaded size:  337 MB
Downloading file................................................................
................................................................................
................................................................................
................................................................................
.............................................
Unpacking...
Traceback (most recent call last):
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 224, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 263, in execute
    output = self.handle(*args, **options)
  File "C:\Python27\lib\site-packages\kalite\distributed\management\commands\set
up.py", line 426, in handle
    call_command("unpack_assessment_zip", ass_item_filename)
  File "C:\Python27\share\kalite\python-packages\django\core\management\__init__
.py", line 161, in call_command
    return klass.execute(*args, **defaults)
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 263, in execute
    output = self.handle(*args, **options)
  File "C:\Python27\lib\site-packages\kalite\contentload\management\commands\unp
ack_assessment_zip.py", line 63, in handle
    zf = zipfile.ZipFile(f, "r")
  File "C:\Python27\lib\zipfile.py", line 770, in __init__
    self._RealGetContents()
  File "C:\Python27\lib\zipfile.py", line 839, in _RealGetContents
    raise BadZipfile("Bad magic number for central directory")
BadZipfile: Bad magic number for central directory

**Comment by MCGallaspy** [comment]
@EdDixon you can enclose things in triple backticks (```) to get formatting like this:
```
Microsoft Windows XP [Version 5.1.2600]
(C) Copyright 1985-2001 Microsoft Corp.

C:\Documents and Settings\IEUser>kalite
C:\Python27\Scripts\kalite:98: RuntimeWarning: Could not determine Python in you
r path, defaulting to C:\Python27\python.exe
  warnings.warn("Could not determine Python in your path, defaulting to {0:s}".f
ormat(python_executable), RuntimeWarning)
Usage:
  kalite start [--foreground] [options] [DJANGO_OPTIONS ...]
  kalite stop [options] [DJANGO_OPTIONS ...]
  kalite restart [options] [DJANGO_OPTIONS ...]
  kalite status [options]
  kalite shell [options] [DJANGO_OPTIONS ...]
  kalite test [options] [DJANGO_OPTIONS ...]
  kalite manage [options] COMMAND [DJANGO_OPTIONS ...]
  kalite -h | --help
  kalite --version

C:\Documents and Settings\IEUser>kalite manage setup
C:\Python27\Scripts\kalite:98: RuntimeWarning: Could not determine Python in you
r path, defaulting to C:\Python27\python.exe
  warnings.warn("Could not determine Python in your path, defaulting to {0:s}".f
ormat(python_executable), RuntimeWarning)

   _   __  ___    _     _ _
  | | / / / _ \  | |   (_) |
  | |/ / / /_\ \ | |    _| |_ ___
  |    \ |  _  | | |   | | __/ _ \
  | |\  \| | | | | |___| | ||  __/
  \_| \_/\_| |_/ \_____/_|\__\___|

https://learningequality.org/ka-lite/

         version 0.14.0

--------------------------------------------------------------------------------

This script will configure the database and prepare it for use.
--------------------------------------------------------------------------------

Press [enter] to continue...
-------------------------------------------------------------------
WARNING: Database file already exists!
-------------------------------------------------------------------
Keep database file and upgrade to KA Lite version 0.14.0? (yes or no) yes
Upgrading database to KA Lite version 0.14.0
Cleaning *pyc files (if writable) from: C:\Program Files\KA Lite\ka-lite\dist-pa
ckages
Cleaning *pyc files (if writable) from: C:\Program Files\KA Lite\ka-lite\python-
packages
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)
Running migrations for securesync:
- Nothing to migrate.
 - Loading initial data for securesync.
Installed 0 object(s) from 0 fixture(s)
Running migrations for config:
- Nothing to migrate.
 - Loading initial data for config.
Installed 0 object(s) from 0 fixture(s)
Running migrations for chronograph:
- Nothing to migrate.
 - Loading initial data for chronograph.
Installed 0 object(s) from 0 fixture(s)
Running migrations for main:
- Nothing to migrate.
 - Loading initial data for main.
Installed 0 object(s) from 0 fixture(s)
Running migrations for playlist:
- Nothing to migrate.
 - Loading initial data for playlist.
Installed 0 object(s) from 0 fixture(s)
Running migrations for updates:
- Nothing to migrate.
 - Loading initial data for updates.
Installed 0 object(s) from 0 fixture(s)
Running migrations for facility:
- Nothing to migrate.
 - Loading initial data for facility.
Installed 0 object(s) from 0 fixture(s)
Running migrations for student_testing:
- Nothing to migrate.
 - Loading initial data for student_testing.
Installed 0 object(s) from 0 fixture(s)
Running migrations for store:
- Nothing to migrate.
 - Loading initial data for store.
Installed 0 object(s) from 0 fixture(s)
Running migrations for i18n:
- Nothing to migrate.
 - Loading initial data for i18n.
Installed 0 object(s) from 0 fixture(s)
Running migrations for tastypie:
- Nothing to migrate.
 - Loading initial data for tastypie.
Installed 0 object(s) from 0 fixture(s)
Syncing...
Creating tables ...
Installing custom SQL ...
Installing indexes ...
Installed 0 object(s) from 0 fixture(s)

Synced:
 > django.contrib.auth
 > django.contrib.sessions
 > django.contrib.messages
 > django.contrib.admin
 > django.contrib.staticfiles
 > django.contrib.contenttypes
 > compressor
 > django_js_reverse
 > south
 > fle_utils.testing
 > kalite.coachreports
 > kalite.distributed
 > kalite.topic_tools
 > kalite.contentload
 > kalite.dynamic_assets
 > kalite.control_panel

Not synced (use migrations):
 - securesync
 - fle_utils.config
 - fle_utils.chronograph
 - kalite.main
 - kalite.playlist
 - kalite.updates
 - kalite.facility
 - kalite.student_testing
 - kalite.store
 - kalite.i18n
 - tastypie
(use ./manage.py migrate to migrate these)

Starting in version 0.13, you will need an assessment items package in order to
access many of the available exercises.
If you have an internet connection, you can download the needed package. Warning
: this may take a long time!
If you have already downloaded the assessment items package, you can specify the
 file in the next step.
Otherwise, we will download it from https://learningequality.org/downloads/ka-li
te/0.14/content/assessment.zip.
Do you wish to download the assessment items package now? (yes or no) yes
Downloading assessment item data from a remote server. Please be patient; this f
ile is big, so this may take some time...
Downloaded size:  337 MB
Downloading file................................................................
................................................................................
................................................................................
................................................................................
.............................................
Unpacking...
Traceback (most recent call last):
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 224, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 263, in execute
    output = self.handle(*args, **options)
  File "C:\Python27\lib\site-packages\kalite\distributed\management\commands\set
up.py", line 426, in handle
    call_command("unpack_assessment_zip", ass_item_filename)
  File "C:\Python27\share\kalite\python-packages\django\core\management\__init__
.py", line 161, in call_command
    return klass.execute(*args, **defaults)
  File "C:\Python27\share\kalite\python-packages\django\core\management\base.py"
, line 263, in execute
    output = self.handle(*args, **options)
  File "C:\Python27\lib\site-packages\kalite\contentload\management\commands\unp
ack_assessment_zip.py", line 63, in handle
    zf = zipfile.ZipFile(f, "r")
  File "C:\Python27\lib\zipfile.py", line 770, in __init__
    self._RealGetContents()
  File "C:\Python27\lib\zipfile.py", line 839, in _RealGetContents
    raise BadZipfile("Bad magic number for central directory")
BadZipfile: Bad magic number for central directory
```

**Comment by MCGallaspy** [comment]
Provisional milestone. @benjaoming do you have any insight here?

**Comment by benjaoming** [comment]
@EdDixon given that it says 0.14.0, is this an installation running from a fresh 0.14.x branch? I would recommend using the alpha releases available through pip or the .debs in the PPA.

That aside (because it's likely not the cause), I think the `BadZipfile: Bad magic number for central directory` could be linked to the re-encoded assessment.zip -- we found that the original zip was 450 MB and with zero compression so we re-compressed and re-uploaded a 330 MB version, maybe @MCGallaspy could you try verifying that the zip (probably created with the ubuntu `zip` program) works ?

@EdDixon could you try re-fetching the .zip in case it was a broken download?

**Comment by benjaoming** [comment]
Also:

> In all installers tested over the past several weeks a large number of assessment items seem to be missing.

There WAS a problem and that one was fixed in #4122 which is not yet merged, so you need to use the latest alpha11 #4140  @EdDixon

**Comment by MCGallaspy** [comment]
@benjaoming he's using the windows installer, which I built from the 0.14.x branch. In particular, since #4122 isn't merged in, then there's no expectation that this issue will be fixed. I'll mark this as resolved when it's merged.

**Comment by EdDixon** [comment]
Awesome!

**Comment by MCGallaspy** [comment]
This doesn't appear to be fixed by #4122.

**Comment by benjaoming** [comment]
No that wasn't what I was saying... I said:

> That aside (because it's likely not the cause), I think the `BadZipfile: Bad magic number for central directory` could be linked to the re-encoded assessment.zip -- we found that the original zip was 450 MB and with zero compression so we re-compressed and re-uploaded a 330 MB version, maybe @MCGallaspy could you try verifying that the zip (probably created with the ubuntu zip program) works ?
